client/server/middleware/assets.js

```diff
@@ -22,9 +22,13 @@ const groupAssetsByType = ( assets ) => defaults( groupBy( assets, getAssetType
 
 export default () => {
 	let assetsFile;
+	let assetsFileModified = 0;
+
 	async function readAssets() {
-		if ( ! assetsFile ) {
+		const stats = await stat( ASSETS_FILE );
+		if ( ! assetsFile || stats.mtimeMs > assetsFileModified ) {
 			assetsFile = JSON.parse( await readFile( ASSETS_FILE, 'utf8' ) );
```

Comment:
@jsnajdr I think the code scanning found something that makes sense here:

![Screenshot 2025-02-05 at 17 16 30](https://github.com/user-attachments/assets/2d60013a-1e40-4db4-945d-617d3a5eed01)

This is theoretically possible. Should we try to circumvent? The check has some suggestions on how to approach it.

Reply:
Nice, I fixed this by using file descriptors (`fd = open()`, `fd.stat()`, `fd.readFile`, `fd.close()`) instead. That's how one would do it in C 🙂

Reply:
Sounds good!